.github/workflows/push_complement_image.yml

```
@@ -55,14 +55,20 @@ jobs:
           password: ${{ secrets.GITHUB_TOKEN }}
       - name: Work out labels for complement image
         id: meta
-        uses: docker/metadata-action@v1
+        uses: docker/metadata-action@v4
         with:
           images: ghcr.io/${{ github.repository }}/complement-synapse
+          tags: |
+            type=schedule,pattern=nightly,enable=${{ github.event_name == 'schedule'}}
+            type=raw,value=develop,enable=${{ github.event_name == 'schedule' || inputs.branch == 'develop' }}
+            type=raw,value=latest,enable=${{ github.event_name == 'push' || inputs.branch == 'master' }}
```

```suggestion
            type=raw,value=develop,enable=${{ github.event_name == 'schedule' && inputs.branch == 'develop' }}
            type=raw,value=latest,enable=${{ github.event_name == 'push' && inputs.branch == 'master' }}
            type=raw,value=latest,enable=${{ github.event_name == 'push' && inputs.branch == 'main' }}
```

I think AND semantics are right here; in case we add more ways this can be run later. Also adding `main` in here to match the docker.yml setup?

---

Or maybe even only care about inputs.branch ?

---

`inputs.branch` is only set when this workflow is triggered via `workflow_dispatch`---it corresponds to lines 12--18. So I think we need the OR here?

---

And the workflow_dispatch is only meant for testing's sake, I wouldn't expect to use it very often once it's working